packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -675,6 +675,10 @@ function isVoidReturningFunctionType(
 function returnsThenable(checker: ts.TypeChecker, node: ts.Node): boolean {
   const type = checker.getApparentType(checker.getTypeAtLocation(node));
 
+  if (type.isUnion()) {
+    return type.types.some(t => anySignatureIsThenableType(checker, node, t));
+  }
+
   if (anySignatureIsThenableType(checker, node, type)) {
     return true;
   }
```

Comment:
```suggestion
  return tsutils.unionTypeParts(type).some(t => anySignatureIsThenableType(checker, node, t));
```

[Refactor] `ts-api-utils` is already imported in this file. What do you think about using the [`unionTypeParts`](https://github.com/JoshuaKGoldberg/ts-api-utils/blob/49d5dd0e1b7f4875ccb9f2e82ddd31706fe4b750/src/types/utilities.ts#L397-L399) utility here?